usr/src/uts/common/fs/zfs/txg.c

```diff
@@ -532,39 +507,50 @@ txg_quiesce_thread(dsl_pool_t *dp)
 	txg_thread_enter(tx, &cpr);
 
 	for (;;) {
-		uint64_t txg;
+		uint64_t delta, timeout, timer, txg;
 
 		/*
-		 * We quiesce when there's someone waiting on us.
-		 * However, we can only have one txg in "quiescing" or
-		 * "quiesced, waiting to sync" state.  So we wait until
-		 * the "quiesced, waiting to sync" txg has been consumed
-		 * by the sync thread.
+		 * Quiesce the current txg whenever there is no txg that has
+		 * quiesced but not yet started syncing, and one of the
+		 * following is true:
+		 *
+		 * (1) Something is waiting for a txg.  This will usually be
+		 *     someone trying to write when their txg is full, or
+		 *     someone trying to push a sync task.
+		 * (2) The txg timeout is hit.  This condition only applies
+		 *     once a quiesced txg enters the sync phase.
```

Comment:
I don't understand what is meant by "This condition only applies once a quiesced txg enters the sync phase."